felix/dataplane/linux/vxlan_mgr.go

```diff
@@ -585,6 +585,28 @@ func (m *vxlanManager) getParentInterface(localVTEP *proto.VXLANTunnelEndpointUp
 	return nil, fmt.Errorf("Unable to find parent interface with address %s", parentDeviceIP)
 }
 
+func (m *vxlanManager) parseMacForIPVersion(vtep *proto.VXLANTunnelEndpointUpdate) (net.HardwareAddr, error) {
+	var (
+		mac net.HardwareAddr
+		err error
+	)
+
+	switch m.ipVersion {
+	case 4:
+		mac, err = net.ParseMAC(vtep.Mac)
```

Comment:
nit, but you could simplify this logic a bit:

```go
switch m.ipVersion {
  case 4:
    return net.ParseMAC(vtep.Mac)
  case 6:
    return net.ParseMAC(vtep.MacV6)
  default:
    return nil, fmt.Errorf(...)
}

Reply:
Thanks @caseydavenport that's much better, will do!